src/amo/components/LandingAddonsCard/index.js

```diff
@@ -35,10 +35,12 @@ export default class LandingAddonsCard extends React.Component {
 
     let footerLinkHtml = null;
     if (addons && addons.length >= placeholderCount) {
-      const linkSearchURL = {
-        ...footerLink,
-        query: convertFiltersToQueryParams(footerLink.query),
-      };
+      const linkSearchURL = typeof footerLink === 'string' ?
```

Comment:
The old component was simply fixing up the `to` property as a convenience for all the component callers. I think this would better encapsulate the intention of fixing `to` properties:

```js
let linkTo = footerLink;
if (typeof linkTo === 'object') {
  // As a convenience, fix the query parameter.
  linkTo = {
    ...linkTo,
    query: convertFiltersToQueryParams(linkTo.query),
  };
}
footerLinkHtml = <Link to={linkTo}>{footerText}</Link>;
```

A better way to fix this whole thing is to make each calling component pass in their own `<Link/>`. Instead of this:

```js
        <LandingAddonsCard
          addons={featuredExtensions}
          className="Home-FeaturedExtensions"
          header={i18n.gettext('Featured extensions')}
          footerText={i18n.gettext('See more featured extensions')}
          footerLink={{
            pathname: '/search/',
            query: {
              addonType: ADDON_TYPE_EXTENSION,
              featured: true,
            },
          }}
          loading={resultsLoaded === false}
        />
```

Pass in a `<Link/>`:

```js
        <LandingAddonsCard
          addons={featuredExtensions}
          className="Home-FeaturedExtensions"
          header={i18n.gettext('Featured extensions')}
          footerText={i18n.gettext('See more featured extensions')}
          footerLink={
            <Link 
              to={{
                pathname: '/search/',
                query: convertFiltersToQueryParams({
                  addonType: ADDON_TYPE_EXTENSION,
                  featured: true,
                }),
              }} 
            />
          }
          loading={resultsLoaded === false}
        />
```

I'm not sure it's worth the effort though -- just something to think about for future designs.

Reply:
Thanks @kumar303. I made the small fix suggested above so we can get this into the tag. I'll merge it once CI passes. I'll open a separate issue about refactoring LandingAddonsCard.